modules/dkan/dkan_datastore/src/DatastoreInterface.php

```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Dkan\Datastore;
+
+interface DatastoreInterface {
+
+  public function __construct($uuid);
+
+  /**
+   * Return the node (resource) associated with this datastore.
+   */
+  public function getNode();
```

Comment:
What do you think about making this more generic - getResource(), perhaps we have option to not store resources in nodes? Implementation of interface can decide how to get the resource... but would probably need a Resource class w/methods to do that

Also - dkan_datastore_resource_load() exists (can't find anything using it though), if you had a Resource class - a loadResourceById() method could do similar to ^^ and check if it's an integer or not so it either loads by NID or UUID. I'm not a big fan of assumed multiple ID formats though.

Reply:
I agree with decoupling this from nodes and only being attached to the more generic concept of "resource" but as you mentioned, that would necessitate some extra abstraction that I did not want to tackle until I knew I could decoupled this thing. I can definitely change the name though and add a @todo for further decoupling later on.